### PR TITLE
Updater: support aarch64 on mac.

### LINKED ISSUE #986: Update LonghornUpdater to specify arch for macOS updates
Right now it looks for assets ending with `-mac.zip`. Starting with `0.7.0` the names will be `-mac.aarch64.zip` or `-mac.x86_64.zip`.

Maybe we need to keep the name for the Intel binaries as `-mac.zip` so that upgrades from `0.6.1` will continue to work?

### PR DESCRIPTION
**Pull request by mook-as**
This adds support for our new mac zip file names (due to the introduction of aarch64 / Apple M1).

This also fixes an issue where if you turn off updates, you can never turn it back on again (because we aborted so early we thought no update server was configured).

This also makes it possible to test updates against GitHub pages acting as the update responder (hard-coding HTTP GET for those).

To test on x86_64 and aarch64:
- Download the appropriate build
- Edit `Rancher Desktop.app/Contents/Resources/app-update.yml` to have the following contents:
   ```yaml
   provider: custom
   upgradeServer: https://mook-as.github.io/rd/response.json
   vPrefixedTagName: true
   owner: mook-as
   repo: rd
   updaterCacheDirName: rancher-desktop-updater
   ```
- Watch the update logs — note that the build on the server is actually broken and doesn't start :)

Fixes #986.

### CODE REVIEW

File: src/main/update/LonghornUpdater.ts [hunk 1]
```diff
@@ -50,8 +52,12 @@ export class MacLonghornUpdater extends MacUpdater {
     return await super.getUpdateInfoAndProvider();
   }
 
+  private readonly isArm64 = Electron.app.runningUnderRosettaTranslation || os.arch() === 'arm64';
+
   findAsset(assets: GithubReleaseAsset[]): GithubReleaseAsset | undefined {
-    return assets.find(asset => asset.name.endsWith('-mac.zip'));
+    const suffix = this.isArm64 ? '-mac.aarch64.zip' : '-mac.x86_64.zip';
```
Comment (ericpromislow) [inline src/main/update/LonghornUpdater.ts:58]: Does this mean that all the new files the upgrade-server delivers end with `-mac.ARCH.zip` ? Are users of older code going to see new versions they can upgrade to, if they're running code that still looks for `*-mac.zip` and none of the files match?  (Of course I don't know exactly what these assets are for...)

**Review by ericpromislow — COMMENTED** [review]
More questions than change requests. Probably ok, but one more cycle pls…

File: src/main/update/LonghornProvider.ts [hunk 2]
```diff
@@ -233,9 +234,11 @@ export default class LonghornProvider extends Provider<UpdateInfo> {
       appVersion: this.updater.currentVersion.format(),
       extraInfo:  { platform: `${ os.platform() }-${ os.arch() }` },
     };
-    const responseRaw = await fetch(
-      this.configuration.upgradeServer,
-      { method: 'POST', body: JSON.stringify(requestPayload) });
+    const requestOptions = /^https?:\/\/[^/]+\.github\.io\//.test(this.configuration.upgradeServer) ? { method: 'GET' } : {
```
[the remark this answers is not in the document]
Comment (mook-as) [reply src/main/update/LonghornProvider.ts:237]: Yes, it's there to support testing with GitHub Pages.

We are using `POST` to pass in params as that's what [longhorn upgrade responder](https://github.com/longhorn/upgrade-responder) expects.  That's not used for GitHub Pages.

File: src/main/update/LonghornUpdater.ts (hunk 1 above)
Comment (mook-as) [reply src/main/update/LonghornUpdater.ts:58]: Yes. No, Matt agreed to manually upload one version with both `-mac.zip` and `-mac.x86_64.zip` (with the same files) to support the upgrade.

Those are just the files you see in GitHub releases pages.

**Review by mook-as — COMMENTED** [review]
Commenting for now, will get to actually making changes later.

File: src/main/update/LonghornProvider.ts (hunk 2 above)
[the remark this answers is not in the document]
Comment (mook-as) [reply src/main/update/LonghornProvider.ts:241]: Comment added in d71c774426226b5f2ec8641d1436a0ae09d2c435.

File: src/main/update/index.ts [hunk 3]
```diff
@@ -117,10 +117,6 @@ mainEvent.on('settings-update', (settings: Settings) => {
  * @returns Whether the update is being installed.
  */
 export default async function setupUpdate(settings: Settings, doInstall = false): Promise<boolean> {
```
[the remark this answers is not in the document]
Comment (mook-as) [reply src/main/update/index.ts]: Cleaned up in 10fcbb75227d56a35d865d240abcbfc010a0b1a3.

**Review by ericpromislow — APPROVED** [review]
The logic involving `checked` and the `autoUpdater` events is a little twisty but it makes sense.

**Pull request merged**